CRM/Contribute/Page/ContributionRecur.php

```diff
@@ -81,21 +88,7 @@ public function view() {
   }
 
   public function preProcess() {
-    $this->_action = CRM_Utils_Request::retrieve('action', 'String', $this, FALSE, 'view');
-    $this->_id = CRM_Utils_Request::retrieve('id', 'Positive', $this);
-    $this->_contactId = CRM_Utils_Request::retrieve('cid', 'Positive', $this, TRUE);
-    $this->assign('contactId', $this->_contactId);
-
-    // check logged in url permission
-    CRM_Contact_Page_View::checkUserPermission($this);
-
-    $this->assign('action', $this->_action);
-
-    if ($this->_permission == CRM_Core_Permission::EDIT && !CRM_Core_Permission::check('edit contributions')) {
```

Comment:
This isn't in the parent function that I can see as yet....

Reply:
I have to say I can't see where 
```
      $this->assign('permission', 'view');
```

would be picked up - or even ```$this->_permission``` - perhaps that code just does nothing anyway?

Reply:
@artfulrobot Re `$this->_action` there is some *magic* in `CRM_Utils_Request::retrieve` that converts from string to integer. That has caught me out before but if you pass a string for action to the function it comes back as an integer!

Re. the permission code, I'm pretty sure the existing check (that I've removed) did nothing sensible. You only *need* view permission to view a recurring contribution, and if you have edit you need view anyway.